core/src/main/java/ai/timefold/solver/core/impl/score/stream/bavet/common/index/ComparisonIndexer.java

```diff
@@ -14,19 +14,37 @@
 final class ComparisonIndexer<T, Key_ extends Comparable<Key_>>
         implements Indexer<T> {
 
-    private final int propertyIndex;
+    private final KeyRetriever<Key_> keyRetriever;
     private final Supplier<Indexer<T>> downstreamIndexerSupplier;
     private final Comparator<Key_> keyComparator;
     private final boolean hasOrEquals;
     private final NavigableMap<Key_, Indexer<T>> comparisonMap;
 
-    public ComparisonIndexer(JoinerType comparisonJoinerType, Supplier<Indexer<T>> downstreamIndexerSupplier) {
-        this(comparisonJoinerType, 0, downstreamIndexerSupplier);
+    /**
+     * Construct an {@link ComparisonIndexer} which immediately ends in a {@link NoneIndexer}.
```

Comment:
The name `NoneIndexer` seems off to me. Maybe should choose a more meaningful name, such as `RawDataIndexer` or something similar.

Reply:
I agree it looks weird, but it matches with `NoneJoiner`. In the absence of better names that would work for both, I'll leave it as is. It's a private API anyway - if a better idea comes along, we can change it any time we like.